packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -480,12 +493,36 @@ class LoadTestDataStore extends DataObject implements ILoadTest {
             }
             return !this.runtime.disposed;
         } finally {
-            if (t !== undefined) {
-                clearTimeout(t);
+            if (timeout !== undefined) {
+                clearTimeout(timeout);
             }
             dataModel.printStatus();
         }
     }
+
+    async sendSignals(config: IRunConfig, timeout: NodeJS.Timeout | undefined) {
+        const clientSignalsSendCount = (typeof config.testConfig.totalSignalsSendCount === "undefined") ?
+                                        0 : config.testConfig.totalSignalsSendCount / config.testConfig.numClients;
+        const cycleMs = config.testConfig.readWriteCycleMs;
+        const signalsPerCycle = (typeof config.testConfig.signalsPerMin === "undefined") ?
+                                 0 : config.testConfig.signalsPerMin * cycleMs / 60000;
+        const signalsGapMs = cycleMs / signalsPerCycle;
+        let submittedSignals = 0;
+        try {
+            while (submittedSignals < clientSignalsSendCount) {
+                // all the clients are sending signals;
```

Comment:
we primarily do this in case there is throttling, so the client has a rest period to recover. do we not throttle on signals? @GaryWilber  do you know?

Reply:
We didn't notice such throttling on signals during our test runs - so I think it may not be necessary.

Reply:
There is also throttling for signals and the throttle limit is higher since signals are cheaper for the system